lib/evmone/symbolic.hpp

```diff
@@ -0,0 +1,254 @@
+#pragma once
+
+#include "baseline.hpp"
+#include "rc_ptr.hpp"
+#include <intx/intx.hpp>
+#include <map>
+#include <variant>
+
+using intx::uint256;
+
+template <class... Ts>
+struct Cases : Ts...
+{
+    using Ts::operator()...;
+};
+
+namespace evmone
+{
+
+struct Pure;
+template<typename>
+struct Load;
+struct UnaryOp;
+struct BinaryOp;
+struct TernaryOp;
+template<typename>
+struct SymbolicUpdates;
+
+template <class T>
+using SymbolicUpdatesPtr = SymbolicUpdates<T>*;
+
+struct SetItem;
+struct Offset;
+struct SetMem;
+
+
+using SymbolicMemoryUpdate =
+    std::variant<SetItem, SetMem, Offset>;
+std::ostream& operator<<(std::ostream& os, const SymbolicMemoryUpdate& i);
+
+
+
+using SymbolicStorage = SymbolicUpdates<SetItem>;
+using SymbolicStoragePtr = SymbolicUpdatesPtr<SetItem>;
+
+
+struct MapComparator
+{
+    bool operator()( const evmc_address& a, const evmc_address& b ) const 
+    {
+        for (size_t i = 0; i < 20; i++)
+        {
+            if(a.bytes[i]<b.bytes[i]) return true;
+        }
+        return false;
```

Comment:
I think this will work:
```suggestion
        if ((uint64_t)a.bytes[0] < (uint64_t)b.bytes[0]) return true;
        if ((uint64_t)a.bytes[8] < (uint64_t)b.bytes[8]) return true;
        return (uint32_t)a.bytes[16] < (uint32_t)b.bytes[16];
```